rewrite-java/src/main/java/org/openrewrite/java/MethodMatcherOption.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2023 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java;
+
+import lombok.Value;
+import org.openrewrite.Incubating;
+import org.openrewrite.Option;
+import org.openrewrite.internal.lang.Nullable;
+
+@Value
+@Incubating(since = "8.12.0")
+public class MethodMatcherOption {
+
+    @Option(displayName = "Method pattern",
+            description = "A method pattern that is used to find matching method declarations/invocations.",
+            example = "org.mockito.Matchers anyVararg()")
+    String methodPattern;
+
+    @Option(displayName = "Match on overrides",
+            description = "When enabled, find methods that are overrides of the method pattern.",
+            required = false)
+    @Nullable
+    Boolean matchOverrides;
+
+    public MethodMatcher methodMatcher() {
+        return new MethodMatcher(methodPattern, matchOverrides != null && matchOverrides);
+    }
+}
```

Comment:
IMHO this is the best example of how this option can be used. I'd want to verify that this works with the SaaS as I expect it to before refactoring all of the existing recipes to use this instead. But for now, this demonstrates a solid example of how this feature can be used.